### DISCUSSION TITLE
How you mock database when creating a unittest cases?

### DISCUSSION THREAD

**Discussion by Naganathan** [kgrzybek/modular-monolith-with-ddd#333] [Q&A]
are you using same database (whatever used in source) for unit test case also? 
 Because it thrown exception, user don't have permission to access for that table.

How are you getting connectionstring from environment variable to mock database in unittest case?
https://github.com/kgrzybek/modular-monolith-with-ddd/blob/master/src/Tests/IntegrationTests/SeedWork/TestBase.cs#L44

Can we use In-Memory sql server database to mock?

**Comment by kacey90** [comment]
Thanks to Testcontainers. You no longer need to mock a db for your unit or integration tests. Lookup [testcontainers](https://testcontainers.com/guides/getting-started-with-testcontainers-for-dotnet/) and get cracking.

**Reply by fatso83** [reply]
Testcontainers are not really fixing any performance issues. It just relieves you of 
1. Installing a postgres server 
2. Having to figure out how to run all tests in a transaction that is rolled back at the end of each test suite

A dedicated postgres instance starts and runs faster than a docker one, but it's still [not as fast or nice as a Fake](https://asgaut.com/use-of-fakes-for-domain-driven-design-and-fast-feedback-loop/). So I am still creating Fakes for Repositories.
